app/Jobs/FixPolygonOverlapJob.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Http\Middleware\SetAuthenticatedUserForJob;
+use App\Models\DelayedJob;
+use App\Services\PolygonService;
+use Exception;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Http\Response;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Throwable;
+
+class FixPolygonOverlapJob implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+
+    private const STATUS_PENDING = 'pending';
+    private const STATUS_FAILED = 'failed';
+    private const STATUS_SUCCEEDED = 'succeeded';
+
+    protected $polygonService;
+
+    protected $polygonUuids;
+
+    protected $job_uuid;
+
+    public $authUserId;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param array $polygonUuids
+     */
+    public function __construct(array $polygonUuids, int $authUserId)
+    {
+        $this->polygonUuids = $polygonUuids;
+        $this->job_uuid = Str::uuid()->toString();
+        $this->authUserId = $authUserId;
+
+    }
+
+    /**
+     * Get the middleware the job should pass through.
+     *
+     * @return array
+     */
+    public function middleware()
+    {
+        return [new SetAuthenticatedUserForJob];
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+
+        try {
+            DelayedJob::create([
+              'uuid' => $this->job_uuid,
```

Comment:
Creating the UUID should be handled by the HasUuid trait, which needs to be added to DelayedJob. There should be no need to create them manually.

Furthermore, the Delayed job should be created when this Job is created, not when it gets started on the worker box in `handle`. The API should be able to find a row in the DB for this job as soon as the UUID is returned from the original controller; the way this is set up now, it's creating a race condition that could cause a 404 to get returned from the v3 delayed jobs endpoint.

Reply:
Job Created before running te worker now.